operator/pkg/ciliumidentity/reconciler.go

```diff
@@ -0,0 +1,474 @@
+	ciliumIdentity resource.Resource[*cilium_api_v2.CiliumIdentity],
+	ciliumEndpoint resource.Resource[*cilium_api_v2.CiliumEndpoint],
+	ciliumEndpointSlice resource.Resource[*v2alpha1.CiliumEndpointSlice],
+	cesEnabled bool,
+	queueOps queueOperations,
+) *reconciler {
+	logger.Info("Creating CID controller Operator reconciler")
+
+	minIDValue := idpool.ID(identity.GetMinimalAllocationIdentity(option.Config.ClusterID))
+	maxIDValue := idpool.ID(identity.GetMaximumAllocationIdentity(option.Config.ClusterID))
+	idAllocator := basicallocator.NewBasicIDAllocator(minIDValue, maxIDValue)
+
+	nsStore, _ := namespace.Store(ctx)
+	podStore, _ := pod.Store(ctx)
+	cidStore, _ := ciliumIdentity.Store(ctx)
+	cepStore, _ := ciliumEndpoint.Store(ctx)
+	cesStore, _ := ciliumEndpointSlice.Store(ctx)
+
+	r := &reconciler{
+		logger:          logger,
+		clientset:       clientset,
+		idAllocator:     idAllocator,
+		desiredCIDState: NewCIDState(logger),
+		cidUsageInPods:  NewCIDUsageInPods(),
+		cidUsageInCES:   NewCIDUsageInCES(),
+		queueOps:        queueOps,
+		nsStore:         nsStore,
+		podStore:        podStore,
+		cidStore:        cidStore,
+		cepStore:        cepStore,
+		cesStore:        cesStore,
+		cesEnabled:      cesEnabled,
+	}
+
+	return r
+}
+
+// syncCESsOnStartup updates the cache of CID usage in CES for all the
+// existing CESs.
+func (r *reconciler) calcDesiredStateOnStartup() error {
+	r.syncCESsOnStartup()
+	return r.syncPodsOnStartup()
+}
+
+func (r *reconciler) syncCESsOnStartup() {
+	if !r.cesEnabled {
+		return
+	}
+
+	for _, ces := range r.cesStore.List() {
+		r.cidUsageInCES.ProcessCESUpsert(ces.Name, ces.Endpoints)
+	}
+}
+
+// syncPodsOnStartup ensures that all pods have a CID for their labels, and that
+// all non-used CIDs are deleted. Non used CIDs are those that aren't in use by
+// any of the pods and also don't exist in CESs (if CES is enabled).
+func (r *reconciler) syncPodsOnStartup() error {
+	var lastError error
+
+	for _, pod := range r.podStore.List() {
+		if err := r.reconcilePod(podResourceKey(pod.Name, pod.Namespace)); err != nil {
+			lastError = err
+		}
+	}
+
+	return lastError
+}
+
+// reconcileCID ensures that the desired state for the CID is reached, by
+// comparing the CID in desired state cache and watcher's store and doing one of
+// the following:
+// 1. Nothing - If CID doesn't exist in both desired state cache and watcher's
+// store.
+// 2. Deletes CID - If CID only exists in the watcher's store, and it isn't used.
+// 3. Creates CID - If CID only exists in the desired state cache.
+// 4. Updates CID - If CIDs in the desired state cache and watcher's store are
+// not the same.
+func (r *reconciler) reconcileCID(cidResourceKey resource.Key) error {
+	cidName := cidResourceKey.Name
+	storeCIDObj, existsInStore, err := r.cidStore.CacheStore().GetByKey(cidResourceKey.Name)
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	var storeCID *cilium_api_v2.CiliumIdentity
+	if existsInStore {
```

Comment:
if is no longer required if we are not using CacheStore()

Reply:
I think that the function has not improved much removing the usage of the `CacheStore()`. Moreover, calling `handleCIDDeletion` is pointless, as in the current implementation, there is another cell taking care of the identities GC. Finally, the `handleCIDDeletion` placeholder should be removed and, in case, it will be added later in PR https://github.com/cilium/cilium/pull/33380 (and possible follow-ups).

Reply:
removed the `handleCIDDeletion`